### PR TITLE
Misc Maintenance

### PR DESCRIPTION
**Pull request by apmac-io**
* Use Logger.warning/2
* Upgrade deps
* Fix test failure using otp-26
* Drop support for Elixir v1.10, v.1.11, v.1.12

### CODE REVIEW

File: test/uniform/app_test.exs [hunk 1]
```diff
@@ -83,19 +83,19 @@ defmodule Uniform.AppTest do
     assert %MixDep{mix_deps: [:indirect_mix]} = app.internal.deps.mix.included_mix
     assert %MixDep{mix_deps: []} = app.internal.deps.mix.indirect_mix
 
-    assert app.internal.deps.included.lib == [
+    assert Enum.sort(app.internal.deps.included.lib) == [
              :always_included_lib,
              :included_lib,
              :indirectly_included_lib
            ]
```
Comment (paulstatezny) [inline test/uniform/app_test.exs:90]: So these were the test failures that broke on OTP 26?

Do you know what underlying behavior changed to cause these behaviors? Just curious.

Comment (apmac-io) [reply test/uniform/app_test.exs:90]: > When migrating to Erlang/OTP 26, keep it mind it changes how maps are stored internally and they will be printed and traversed in a different order... [Elixir v1.14.4 release notes](https://github.com/elixir-lang/elixir/releases/tag/v1.14.4)

Remember how maps used to have a stable sort order if the number of items was `n` (32, right) or less?. OTP 26 changes this, so if you have a map that gets converted into a list at some point, the resulting order of those items is non-deterministic.

File: test/projects/full/mix.exs [hunk 2]
```diff
@@ -25,8 +25,7 @@ defmodule Full.MixProject do
       # comment to remove
       {:esbuild, "~> 0.5", runtime: Mix.env() == :dev},
       {:graphql_document, "~> 0.2.1"},
-      # test that options are retained
-      {:sourceror, "~> 0.11.1", path: "../../../deps/sourceror", override: true},
+      {:sourceror, "~> 0.14"},
```
Comment (apmac-io) [inline test/projects/full/mix.exs:28]: 🤔 Uncertain how to preserve this...or if even we need to...?

Comment (capitalist) [reply test/projects/full/mix.exs:28]: I ejected a complex app and compare the before/after and the results are identical.

**Review by capitalist — APPROVED** [review]
👍

**Pull request merged**